salt/modules/archive.py

```diff
@@ -168,7 +172,15 @@ def _list_tar(name, cached, decompress_cmd, failhard=False):
             return dirs, files, links
 
         except tarfile.ReadError:
-            if not failhard:
+            if failhard:
+                if fileobj:
```

Comment:
Yeah, that was leftover from earlier. I had initially added a ``fileobj`` arg to ``_list_tar()`` and then decided to remove it and instead just check if ``cached`` was a Popen object. I just didn't catch this other reference. Thanks!